batch/batch/cloud/gcp/driver/create_instance.py

```diff
@@ -43,7 +43,8 @@ def create_vm_config(
     _, cores = gcp_machine_type_to_worker_type_and_cores(machine_type)
 
     region = instance_config.region_for(zone)
-
+    nvidia_runtime = "--runtime=nvidia" if 'g2' in machine_type else ""
+    gpus_param = "--gpus all" if 'g2' in machine_type else ""
```

Comment:
I think it will be useful to add a function (in batch/cloud/gcp/resource_utils.py, say) that returns whether a given machine_type has GPUs.  Then, when we add support for another machine type (like a2s), we only need to change it in one place.

You can also combine these options into a single variable, maybe something like `docker_run_gpu_args`.